src/lib/settings/gateway/GatewayStorage.js

```diff
@@ -46,9 +48,11 @@ class GatewayStorage {
 
 		/**
 		 * @since 0.5.0
+		 * @name GatewayStorage#schema
 		 * @type {Schema}
+		 * @readonly
 		 */
-		this.schema = schema;
+		Object.defineProperty(this, 'schema', { value: schema || new Schema(), enumerable: true });
```

Comment:
We should default these in constants and use mergeDefault just like every other file. Also in the event we add more things that might need to be defaulted